pkg/clustermesh/mcsapi/serviceimport_controller.go

```diff
@@ -416,6 +454,15 @@ func (r *mcsAPIServiceImportReconciler) Reconcile(ctx context.Context, req ctrl.
 	svcImport.Spec.Type = oldestClusterSvc.Type
 	svcImport.Spec.SessionAffinity = oldestClusterSvc.SessionAffinity
 	svcImport.Spec.SessionAffinityConfig = oldestClusterSvc.SessionAffinityConfig.DeepCopy()
+	svcImport.Labels = maps.Clone(oldestClusterSvc.Labels)
+	annotations := maps.Clone(oldestClusterSvc.Annotations)
+	if _, ok := svcImport.Annotations[mcsapicontrollers.DerivedServiceAnnotation]; ok {
+		if annotations == nil {
+			annotations = map[string]string{}
+		}
+		annotations[mcsapicontrollers.DerivedServiceAnnotation] = svcImport.Annotations[mcsapicontrollers.DerivedServiceAnnotation]
+	}
+	svcImport.Annotations = annotations
```

Comment:
Just a note that this would still be problematic if any new label/annotation would be added in the future, as well as it resets any label/annotation that a user could manually add. I don't have great ideas about how to address that though, as you'd need to either keep track of the previously added labels/annotations (so that they can be removed afterwards if no longer present in the ServiceExport), or leverage ServerSideApply, which does that for you --- but it doesn't seem to be yet supported in controller runtime). That said, I'm fine with merging as is, and iterating again in the future if need arises.

Reply:
Makes sense! However I am not entirely sure we would add more annotation/labels controlled this way. We have some for the derived service but it would probably be unlikely for the ServiceImport resource, or at least I don't see the case where we would need it now :sweat_smile: